src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -3,557 +3,358 @@
-   * to be implemented by the client and can be dangerous if done incorrectly.
+   * <p>This write forces the client to go over its maxBufferSize.  This can be dangerous if used incorrectly.
+   * Its assumed if this is used you are keeping track of the clients writeBuffer on your own.</p>
    * 
+   * <p>As this write is thread safe it should be noted that calling this with multiThreads can not garentee order.
+   * As such it is recommended to write full parsable protocol packets at a time, or implement your own locking around the
+   * write if you are streaming data raw data and using more then 1 thread to do such.
+   * </p>
    * 
    * @param bb the ByteBuffer to write to the client.
    */
-  public void writeForce(ByteBuffer bb) {
-    synchronized(writeBuffers) {
-        boolean needNotify = ! canWrite();
-        this.addToWriteSize(bb.remaining());
-        writeBuffers.add(bb.slice());
-        if(needNotify) {
-          this.flagWriteable();
-        }
-    }
-  }
-
+  public void writeForce(ByteBuffer bb);
+  
   /**
-   * This is called by the SocketExecuter once the clients socket is able to be written to.
-   * It is up to the clients implementer to figure out how to deal with combining or not combining ByteBuffers 
-   * for write.  In general its much better to just send the byteBuffers as they where received but the write functions. 
-   * 
-   * @return a ByteBuffer that can be used to Read new data off the socket for this client.
-   */
-  protected ByteBuffer getWriteBuffer() {
-    synchronized(writeBuffers) {
-      //This is to keep from doing a ton of little writes if we can.  We will try to 
-      //do at least 8k at a time, and up to 65k if we are already having to combine buffers
-      if(currentWriteBuffer == null || currentWriteBuffer.remaining() == 0) {
-        if(writeBuffers.nextPopSize() < 65536/8 && writeBuffers.remaining() > writeBuffers.nextPopSize()) {
-          if(writeBuffers.remaining() < 65536) {
-            currentWriteBuffer = writeBuffers.pull(writeBuffers.remaining());
-          } else {
-            currentWriteBuffer = writeBuffers.pull(65536);
-          }
-        } else {
-          currentWriteBuffer = writeBuffers.pop();
-        }
-      }
-      return currentWriteBuffer;
-    }
-  }
+   * <p>This provides the next available Write buffer.  This is typically only called by the {@link SocketExecuterInterface}.
+   * This is not threadsafe as the same {@link ByteBuffer} will be provided to any thread that calls this, until 
+   * something consumes all byte available in it at which point another {@link ByteBuffer} will be provided.</p>
+   * 
+   * <p>Assuming something actually removes data from the returned {@link ByteBuffer} an additional call to 
+   * {@link #reduceWrite(int)} must be made</p>
+   * 
+   * 
+   * @return a {@link ByteBuffer} that can be used to Read new data off the socket for this client.
+   */
+  public ByteBuffer getWriteBuffer();
```

Comment:
Wish I could but how do you hide it as an interface??  SocketExecuters only know about Clients/Servers.  As above this was protected when client was a Abstract class.

For this the only option I see is to move back to the very very old behavior of having the SocketExecuter be the one that allocates the BB and .splits it up to the clients.  The only reason I dont like it is because technically you can get access to buffers from other sockets, and someone implemeting it wrong could allow someone else to pull that data.

That said if you think its worth it I can mess around with moving back to that.